src/SetupApi.ts

```diff
@@ -22,7 +22,7 @@ export abstract class SetupApi<EventsMap extends EventMapType> {
 
   public readonly events: LifeCycleEventEmitter<EventsMap>
 
-  constructor(initialHandlers: Array<RequestHandler>) {
+  constructor(...initialHandlers: Array<RequestHandler>) {
```

Comment:
Because the accepted `initialHandlers` were never spread, they always arrive as an Array of handlers, regardless of how they were provided to the `setupServer`/`setupWorker` function calls. 

Preserving the argument spread enables the correct validation.